.github/workflows/release.yml

```diff
@@ -122,82 +122,104 @@ jobs:
+      #     echo "EXPECTED_TRACE_TEMPLATE=adot/utils/expected-templates/java-okhttp-wrapper.json" >> $GITHUB_ENV;
+      - name: Configure NodeJS Values
+        if: ${{ github.event.inputs.layer_name }} == nodejs
+        run: >-
+          echo "SAMPLE_APP_NAME=nodejs-awssdk" >> $GITHUB_ENV;
+          echo "LANGUAGE=nodejs" >> $GITHUB_ENV;
+          echo "BUILD_DIRECTORY=opentelemetry-lambda/nodejs" >> $GITHUB_ENV;
+          echo "BUILD_COMMAND=npm install" >> $GITHUB_ENV;
+          echo "TERRAFORM_DIRECTORY=sample-apps/nodejs-aws-sdk-terraform" >> $GITHUB_ENV;
+          echo "EXPECTED_TRACE_TEMPLATE=adot/utils/expected-templates/nodejs-awssdk.json" >> $GITHUB_ENV;
+      - name: Configure Python Values
+        if: ${{ github.event.inputs.layer_name }} == python38
+        run: >-
+          echo "SAMPLE_APP_NAME=python38" >> $GITHUB_ENV;
+          echo "LANGUAGE=python" >> $GITHUB_ENV;
+          echo "BUILD_DIRECTORY=opentelemetry-lambda/python" >> $GITHUB_ENV;
+          echo "BUILD_COMMAND='cd sample-apps;./build.sh'" >> $GITHUB_ENV;
+          echo "TERRAFORM_DIRECTORY=sample-apps/python-aws-sdk-aiohttp-terraform" >> $GITHUB_ENV;
+          echo "EXPECTED_TRACE_TEMPLATE=adot/utils/expected-templates/python.json" >> $GITHUB_ENV;
+      - name: Configure Collector Values
+        if: ${{ github.event.inputs.layer_name }} == collector
+        run: >-
+          echo "SAMPLE_APP_NAME=dotnet-awssdk-wrapper" >> $GITHUB_ENV;
+          echo "LANGUAGE=dotnet" >> $GITHUB_ENV;
+          echo "BUILD_DIRECTORY=dotnet" >> $GITHUB_ENV;
+          echo "BUILD_COMMAND=./build.sh" >> $GITHUB_ENV;
+          echo "TERRAFORM_DIRECTORY=sample-apps/dotnet-wrapper-aws-sdk-terraform" >> $GITHUB_ENV;
+          echo "EXPECTED_TRACE_TEMPLATE=adot/utils/expected-templates/dotnet-awssdk-wrapper.json" >> $GITHUB_ENV;
+      - name: Set Configuration Outputs
+        id: set-smoke-test-outputs
+        run: >-
+          echo "::set-output name=sample_app_name::$SAMPLE_APP_NAME";
+          echo "::set-output name=language::$LANGUAGE";
+          echo "::set-output name=build_directory::$BUILD_DIRECTORY";
+          echo "::set-output name=terraform_directory::$TERRAFORM_DIRECTORY";
+          echo "::set-output name=expected_trace_template::$EXPECTED_TRACE_TEMPLATE";
+          echo "::set-output name=expected_metric_template::$EXPECTED_METRIC_TEMPLATE";
+          echo "::set-output name=amp_regions::$AMP_REGIONS";
+  smoke-test:
+    name: Smoke Test - ${{ matrix.aws_region }} - ${{ github.event.inputs.layer_name }}
+    needs: get-smoke-test-configuration
+    matrix:
+      include:
+        - aws_region: ${{fromJson(github.event.inputs.aws_region).aws_region}}
     steps:
       - uses: actions/checkout@v2
         with:
           submodules: recursive
       - uses: actions/setup-java@v2
-        if: ${{ matrix.language == 'java' }}
+        if: ${{ needs.get-smoke-test-configuration.outputs.language == 'java' }}
```

Comment:
Why is this prefixed with `needs`?

Reply:
This value is coming from a previous job which it "needs" to begin executing. Now we have 2 jobs: `get-smoke-test-configuration` and `smoke`-test`. As opposed to before where everything was in the same `smoke-test` job.